packages/moonstone/Scrollable/Scrollable.js

```diff
@@ -485,6 +531,55 @@ class ScrollableBase extends Component {
 		this.uiRef.bounds.scrollHeight = this.uiRef.getScrollBounds().scrollHeight;
 	}
 
+	getScrollabilities = () => {
+		if (this.uiRef) {
+			const bounds = this.uiRef.getScrollBounds();
+			return {
+				horizontal: this.uiRef.canScrollHorizontally(bounds),
+				vertical: this.uiRef.canScrollVertically(bounds)
+			};
+		} else {
+			return {horizontal: false, vertical: false};
+		}
+	}
+
+	clearOverscrollEffect = (orientation) => {
+		const
+			getOverscrollStatus = this.uiRef.getOverscrollStatus,
+			status = getOverscrollStatus(orientation);
+
+		if (status > overscrollNeeds.tracking) { // clearing or delayedClearing
+			const timeout = (status === overscrollNeeds.delayedClearing) ? overscrollTimeout : 0;
+			setTimeout(this.playOverscrollEffect, timeout, this.overscrollRefs[orientation], orientation, 'before', 0, false);
+			setTimeout(this.playOverscrollEffect, timeout, this.overscrollRefs[orientation], orientation, 'after', 0, false);
+		}
+	}
+
+	clearAllOverscrollEffects = () => {
+		this.clearOverscrollEffect('horizontal');
+		this.clearOverscrollEffect('vertical');
+	}
+
+	playOverscrollEffect = (nodeRef, orientation, position, ratio, auto) => {
+		const prefix = '--scrollable-overscroll-ratio-';
+
+		nodeRef.style.setProperty(prefix + orientation + position, ratio);
```

Comment:
As of React 15.6.0, we no longer need to use `ref` to set CSS variables. These can be done much more simply by setting `style`, e.g.:

```jsx
<Scrollable
	style={{prefix + orientation + position: ratio}}
```
or a more literal example:
```jsx
<Scrollable
	style={{'--scrollable-overscroll-ratio-vertical-before': ratio}}
```

That should simplify things quite a lot.

Reply:
Those CSS variables are very frequently updated especially on dragging, so I have a concern about the performance impact when we put these into JSX. I don't want to push Scrollable to go into any heavy life cycle.

Reply:
That's a fair point. Though, I don't think they need to be updated at all during scrolling, just once at the very end of scrolling. If it's more than that, it's unnecessary.

Reply:
So the dragging is the key case. :) To be natural, an effect reacts ASAP along with the pointer moving. Scrollable's `componentDidUpdate` has a lot of jobs and it looks not good to invoke an additional React update cycle for a style.